src/vg_protocol.erl

```diff
@@ -118,16 +118,36 @@ encode_record(Record) ->
       Id :: integer(),
       Value :: vg:record(),
       EncodedLog :: {pos_integer(), pos_integer(), iolist()} | {error, bad_checksum}.
+%% if we have an id, we're coming in from write repair and have to handle the checksum differently
+encode_log(Id, #{crc := CRC, id := LogID, record := Record}) ->
+    case LogID of
+        Id ->
+            Record2 = [<<?MAGIC:8/signed-integer, 0:8/signed-integer>>, encode_kv(Record)],
+            case erlang:crc32(Record2) of
+                CRC ->
+                    RecordIoList = [<<CRC:32/unsigned-integer>>, Record2],
+                    RecordSize = erlang:iolist_size(RecordIoList),
+                    {Id+1, RecordSize+12, [<<Id:64/signed-integer, RecordSize:32/signed-integer>>, RecordIoList]};
+                BadChecksum ->
+                    lager:error("checksums don't match crc1=~p crc2=~p", [CRC, BadChecksum]),
+                    {error, bad_checksum}
+            end;
+        _ ->
+            lager:error("ids don't match on write repair id1=~p id2=~p", [Id, LogID]),
+            {error, bad_id}
+    end;
+%% here we're coming from a decoded produce request.
 encode_log(Id, #{crc := CRC, record := Record}) ->
     case erlang:crc32(Record) of
         CRC ->
             RecordIoList = [<<CRC:32/unsigned-integer>>, Record],
             RecordSize = erlang:iolist_size(RecordIoList),
             {Id+1, RecordSize+12, [<<Id:64/signed-integer, RecordSize:32/signed-integer>>, RecordIoList]};
         BadChecksum ->
-            lager:info("checksums don't match crc1=~p crc2=~p", [CRC, BadChecksum]),
+            lager:error("checksums don't match crc1=~p crc2=~p", [CRC, BadChecksum]),
```

Comment:
I prefer to leave `error` for exceptions that we'd want to be alerts of some sort -- not necessarily pagerduty alerts but ones that go to slack or similar. Would this be the case for a bad checksum?

Same for the use of error above in the first encode_log function.

Reply:
I think that either write repair or produce will just fail and need to be retried, but a persistent one might need manual intervention?

Reply:
Then that would be an alert based on the 'retry' metric from prometheus?